kornia/utils/grid.py

```diff
@@ -95,7 +95,7 @@ def create_meshgrid3d(
     if normalized_coordinates:
         xs = (xs / (width - 1) - 0.5) * 2
         ys = (ys / (height - 1) - 0.5) * 2
-        zs = (ys / (height - 1) - 0.5) * 2
+        zs = (zs / (depth - 1) - 0.5) * 2
     # generate grid by stacking coordinates
     base_grid: torch.Tensor = torch.stack(torch.meshgrid([zs, xs, ys])).transpose(1, 2)  # 3xHxW
     return base_grid.unsqueeze(0).permute(0, 3, 4, 2, 1)  # 1xHxWx3
```

Comment:
```suggestion
    return base_grid.unsqueeze(0).permute(0, 3, 4, 2, 1)  # 1xDxHxWx3
```

Reply:
I think also that the transpose/permute could be done just once

Reply:
Also seeing that we don’t have proper tests here. Otherwise this change might have flagged up some errors /cc @shijianjian

Reply:
@edgarriba I added another comment to merge transpose and permute.
However, I don't think what you mentioned here is related to this pull request.